src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -220,51 +221,71 @@ private static async Task<WasmFetchResponse> CallFetch(HttpRequestMessage reques
+                        using (JSObject transformStream = BrowserHttpInterop.CreateTransformStream())
+                        {
+                            Task<JSObject> fetchPromise = BrowserHttpInterop.Fetch(uri, headerNames.ToArray(), headerValues.ToArray(), optionNames, optionValues, abortController, transformStream);
+                            Task<JSObject> fetchTask = BrowserHttpInterop.CancelationHelper(fetchPromise, cancellationToken).AsTask(); // initialize fetch cancellation
+
+                            using (WasmHttpWriteStream stream = new WasmHttpWriteStream(transformStream))
+                            {
+                                try
+                                {
+                                    await request.Content.CopyToAsync(stream, cancellationToken).ConfigureAwait(true);
+                                    Task closePromise = BrowserHttpInterop.TransformStreamClose(transformStream);
+                                    await BrowserHttpInterop.CancelationHelper(closePromise, cancellationToken).ConfigureAwait(true);
+                                }
+                                catch (Exception)
+                                {
+                                    BrowserHttpInterop.TransformStreamAbort(transformStream);
+                                    if (!abortController.IsDisposed)
+                                    {
+                                        BrowserHttpInterop.AbortRequest(abortController);
+                                    }
+                                    try
+                                    {
+                                        using (fetchResponse = await fetchTask.ConfigureAwait(true)) // observe exception
+                                        {
+                                            BrowserHttpInterop.AbortResponse(fetchResponse);
+                                        }
+                                    }
+                                    catch { /* ignore */ }
+                                    cancellationToken.ThrowIfCancellationRequested();
+                                    throw;
+                                }
+                            }
+
+                            fetchResponse = await fetchTask.ConfigureAwait(true);
+                        }
                     }
                     else
                     {
                         byte[] buffer = await request.Content.ReadAsByteArrayAsync(cancellationToken).ConfigureAwait(true);
                         cancellationToken.ThrowIfCancellationRequested();
 
-                        promise = BrowserHttpInterop.Fetch(uri, headerNames.ToArray(), headerValues.ToArray(), optionNames, optionValues, abortController, buffer);
+                        Task<JSObject> fetchPromise = BrowserHttpInterop.Fetch(uri, headerNames.ToArray(), headerValues.ToArray(), optionNames, optionValues, abortController, buffer);
```

Comment:
did we lose ability to abort request here ? I mean, not just abandon the promise, but to stop wasting bandwidth.

Reply:
No, we still do abort the fetch with the `abortController`.
See https://github.com/dotnet/runtime/pull/91699/files#diff-66ce677ded58c3a6a0639d321f4460bb0b2c4bcaeb3216ff3db4381937365bd8R136